### PR TITLE
feat: detected object validation

### PR DESCRIPTION
**Pull request by yukkysaito**
Signed-off-by: Yukihiro Saito <yukky.saito@gmail.com>

## Description(required)
please see [here](https://github.com/autowarefoundation/autoware.universe/blob/eb3a3bf1ec2b5641245874fb4eac2d64c51a6c6c/perception/detected_object_validation/README.md)

image (7)

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Read [commit-guidelines][commit-guide

- [x] Assign PR to reviewer

If you are adding new package following items are required:

- [ ] Documentation with description of the package is available
- [ ] A sample launch file and parameter file are available if the package contains executable nodes

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [x] Commits are properly organized and messages are according to the guideline
- [x] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[commit-guidelines]: https://www.conventionalcommits.org/en/v1.0.0/
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

**Comment by yukkysaito** [comment]
The launch is a separate PR launch.

**Review by miursh — COMMENTED** [review]
Looks almost good.
I left some minor comment and questions.

File: perception/detected_object_validation/include/occupancy_grid_based_validator/occupancy_grid_based_validator.hpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
+// Copyright 2022 Tier IV, Inc.
```
[the remark this answers is not in the document]
Comment (yukkysaito) [reply perception/detected_object_validation/include/occupancy_grid_based_validator/occupancy_grid_based_validator.hpp:1]: I don't know whether logo or the name of the company.
other source codes kept it Tier IV

File: perception/detected_object_validation/launch/occupancy_grid_based_validator.launch.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+
+  <node pkg="laserscan_to_occupancy_grid_map" exec="laserscan_to_occupancy_grid_map_node" name="single_frame_laserscan_occupancy_grid_map" output="screen">
+    <remap from="~/input/laserscan" to="/perception/occupancy_grid_map/virtual_scan/laserscan" />
+    <remap from="~/output/occupancy_grid_map" to="$(var input/occupancy_grid_map)" />
+    <param name="input_obstacle_pointcloud" value="false" />
+    <param name="input_obstacle_and_raw_pointcloud" value="false" />
+    <param name="map_resolution" value="0.2" />
+    <param name="map_length" value="200.0" />
+    <param name="map_length" value="20.0" />
```
[the remark this answers is not in the document]
Comment (yukkysaito) [reply perception/detected_object_validation/launch/occupancy_grid_based_validator.launch.xml]: [ac74d88](https://github.com/autowarefoundation/autoware.universe/pull/507/commits/ac74d885b48f7158a59807558735e3236fef7146)

File: perception/detected_object_validation/src/occupancy_grid_based_validator.cpp [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+{
+boost::optional<geometry_msgs::msg::Transform> getTransform(
+  const tf2_ros::Buffer & tf_buffer, const std::string & source_frame_id,
+  const std::string & target_frame_id, const rclcpp::Time & time)
+{
+  try {
+    geometry_msgs::msg::TransformStamped self_transform_stamped;
+    self_transform_stamped = tf_buffer.lookupTransform(
+      /*target*/ target_frame_id, /*src*/ source_frame_id, time,
+      rclcpp::Duration::from_seconds(0.5));
```
[the remark this answers is not in the document]
Comment (yukkysaito) [reply perception/detected_object_validation/src/occupancy_grid_based_validator.cpp:37]: There is no particular reason here, but it is related stability of execution time. The theoretical value should be 0.1. In reality, it varies up to about 0.3. Therefore, 0.5 is used including margin.

File: perception/detected_object_validation/launch/occupancy_grid_based_validator.launch.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (miursh) [reply perception/detected_object_validation/launch/occupancy_grid_based_validator.launch.xml]: map_width=20 makes output weird like this.
image

[the remark this answers is not in the document]
Comment (yukkysaito) [reply perception/detected_object_validation/launch/occupancy_grid_based_validator.launch.xml]: image
sorry. I fixed and comfirmed

[the remark this answers is not in the document]
Comment (yukkysaito) [reply perception/detected_object_validation/launch/occupancy_grid_based_validator.launch.xml]: @miursh [e3aa520](https://github.com/autowarefoundation/autoware.universe/pull/507/commits/e3aa52042741bc68da5bf6d1fe8bcb3bf5217fa6)

**Review by miursh — APPROVED** [review]
LGTM

**Pull request merged**